python/lsst/ap/verify/metrics.py

```diff
@@ -76,16 +76,31 @@ class MetricsParser(argparse.ArgumentParser):
     def __init__(self):
         # Help and documentation will be handled by main program's parser
         argparse.ArgumentParser.__init__(self, add_help=False)
-        self.add_argument('--metrics-file', default='ap_verify.verify.json',
-                          help='The file to which to output metrics in lsst.verify format. '
-                               'Defaults to ap_verify.verify.json.')
+        self.add_argument(
+            '--metrics-file', default='ap_verify.{dataId}.verify.json',
+            help="The file template to which to output metrics in lsst.verify "
+                 "format; {dataId} will be replaced with the job\'s data ID. "
+                 "Defaults to ap_verify.{dataId}.verify.json.")
         self.add_argument('--silent', dest='submitMetrics', action='store_false',
                           help='Do NOT submit metrics to SQuaSH (not yet implemented).')
         # Config info we don't want on the command line
         self.set_defaults(user=os.getenv(_ENV_USER), password=os.getenv(_ENV_PASSWORD),
                           squashUrl=os.getenv(_ENV_URL, _SQUASH_DEFAULT_URL))
 
 
+# borrowed from validate_drp
+def _extract_instrument_from_butler(butler):
+    """Extract the last part of the mapper name from a Butler repo.
+    'lsst.obs.lsstSim.lsstSimMapper.LsstSimMapper' -> 'LSSTSIM'
+    'lsst.obs.cfht.megacamMapper.MegacamMapper' -> 'CFHT'
+    'lsst.obs.decam.decamMapper.DecamMapper' -> 'DECAM'
+    'lsst.obs.hsc.hscMapper.HscMapper' -> 'HSC'
```

Comment:
Might be worth pointing out these are examples and it will still return an uppercase instrument name even if it's not in this list (for way in the future when our code works with every telescope ever)